src/KeyVault/KeyVault/Models/KeyVaultCmdletBase.cs

```diff
@@ -85,6 +86,22 @@ protected string GetDefaultFileForOperation(string operationName, string vaultNa
             return filename;
         }
 
+        /// <summary>
+        /// Utility function that will split the Vault secret uri into
+        /// `vaultName`, `secretName`, `Version`
+        /// </summary>
+        /// <param name="secretUri">The user's input unique resource identifier</param>
+        /// <returns>An instance of SecretUriComponents</returns>
+        protected SecretUriComponents SplitSecretUri(Uri secretUri)
+        {
+            var vaultName = secretUri.Host.Split('.')[0]; // Extract vault name from the URI
+            var secretName = secretUri.Segments.Length > 2 ? secretUri.Segments[2].TrimEnd('/') : string.Empty; // Extract secret name from the URI
+            var secretVersion = secretUri.Segments.Length > 3 ? secretUri.Segments[3] : string.Empty; // Extract secret version (if present)
```

Comment:
- can we let the parameter as string?
- what happens if secretUri or secretName is empt?

Reply:
Yes, i can make them strings.
If they are empty, they wont be used for any execution logic.

Reply:
any thoughts if we add a constructor for SecretUriComponents and its input a string

Reply:
Yes this will be a much cleaner way to implement this! I have implemented the changes!